### PR TITLE
Config List in Menu

### PR DESCRIPTION
**Pull request by LightPower392**
This pull request adds a way to create, load, and delete configs through the menu. 
config
There are three text inputs where you enter the name of a config (no .conf). After pressing enter, the box automatically becomes empty. If the save or delete are used, then the list will update to reflect the change.

### CODE REVIEW

**Comment by TotallyNotElite** [comment]
Same criticism as in the issue.

**Comment by LightPower392** [comment]
The reason why there are so many commits for this pull request is because I branched the tooltip fork. Could you tell me what the alternative to an input field is?

**Comment by TotallyNotElite** [comment]
Do you mind if I push to your branch, so I can fix this commit mess?

**Comment by LightPower392** [comment]
Sure

**Comment by TotallyNotElite** [comment]
There we go.

File: include/visual/menu/menu/special/ConfigsManagerList.hpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    void recursiveWork(TreeNode &node, size_t depth);
+
+    void construct();
+
+    void addVariable(std::string name, bool registered);
+
+    void removeVariables();
+
+
+    //
```
Comment (BenCat07) [inline include/visual/menu/menu/special/ConfigsManagerList.hpp:41]: \\

File: src/settings/SettingCommands.cpp [hunk 2]
```diff
@@ -34,25 +50,25 @@ static std::string getAutoSaveConfigPath()
     time_info = localtime(&current_time);
     strftime(timeString, sizeof(timeString), "%Y-%m-%d %H:%M", time_info);
 
-    DIR *config_directory = opendir(paths::getConfigPath().c_str());
+    DIR *config_directory = opendir(DATA_PATH "/configs");
```
Comment (BenCat07) [inline src/settings/SettingCommands.cpp:53]: This should continue using paths::getConfigPath()

File: src/visual/menu/GuiInterface.cpp [hunk 3]
```diff
@@ -119,7 +120,7 @@ static void initPlayerlist()
 
 static void load()
 {
-    zerokernel::Menu::instance->loadFromFile(paths::getDataPath("/menu"), "menu.xml");
```
Comment (BenCat07) [inline src/visual/menu/GuiInterface.cpp:122]: Same here for DataPath, and everywhere else paths:: was deleted

**Pull request closed without merging**